code/modules/surgery/organs/organ_external.dm

```diff
@@ -234,34 +258,59 @@
 
 	if(owner_old)
 		owner_old.updatehealth("limb receive damage")
+		if(stamina)
+			owner_old.update_stamina()
 	return update_icon()
 
 #undef LIMB_SHARP_THRESH_INT_DMG
 #undef LIMB_THRESH_INT_DMG
 #undef LIMB_DMG_PROB
 
-/obj/item/organ/external/proc/heal_damage(brute, burn, internal = 0, robo_repair = 0, updating_health = TRUE)
+/obj/item/organ/external/proc/heal_damage(brute, burn, stamina, internal = 0, robo_repair = 0, updating_health = TRUE)
 	if(is_robotic() && !robo_repair)
 		return
 
 	brute_dam = max(brute_dam - brute, 0)
 	burn_dam  = max(burn_dam - burn, 0)
+	stamina_dam = max(stamina_dam - stamina, 0)
 
 	if(internal)
 		status &= ~ORGAN_BROKEN
 		perma_injury = 0
 
 	if(updating_health)
 		owner.updatehealth("limb heal damage")
-
+	consider_processing()
+	check_disabled()
 	return update_icon()
 
+/obj/item/organ/external/proc/total_damage(include_stamina = FALSE)
```

Comment:
renamed proc to total_damage due to having our own get_damage proc that uses a perma_injury var. let me know if you would rather I remove the old get_damage proc